components/core/tests/test-NetworkReader.cpp

```diff
@@ -188,3 +190,53 @@ TEST_CASE("network_reader_illegal_offset", "[NetworkReader]") {
     size_t pos{};
     REQUIRE((clp::ErrorCode_Failure == reader.try_get_pos(pos)));
 }
+
+TEST_CASE("network_reader_with_custom_headers", "[NetworkReader]") {
+    std::map<std::string, std::string> custom_headers = std::map<std::string, std::string>();
+    // We use httpbin (https://httpbin.org/) to test the custom headers. This request will return a
+    // JSON object that contains the custom headers. We check if the headers are in the response.
+    constexpr int kNumRegularTestHeaders = 10;
+    for (int i = 0; i < kNumRegularTestHeaders; i++) {
+        std::string key = fmt::format("Unit-Test-Key{}", i);
+        std::string value = fmt::format("Unit-Test-Value{}", i);
+        custom_headers[key] = value;
+    }
```

Comment:
```suggestion
constexpr size_t cNumRegularTestHeaders{10};
    for (size_t i{0}; i < cNumRegularTestHeaders; ++i) {
        custom_headers.emplace(
                fmt::format("Unit-Test-Key{}", i),
                fmt::format("Unit-Test-Value{}", i)
        );
    }
```
1. We should use `size_t` instead of `int` for general integer use
2. We should use `{}` instead of `=` for variable initialization
3. We should avoid using `[]` for map/unordered_map data insertion/access
4. Use `emplace` with rvalue directly to exercise inline construction

Reply:
fixed in next commit.